src/pages/search.tsx

```diff
@@ -1,67 +1,157 @@
 import { useQuery } from "@apollo/client";
-import { useRouter } from "next/router";
 import React, { useContext, useState } from "react";
 import { useMediaQuery } from "react-responsive";
 import PageWithNavAndPanel from "../components/PageWithNavAndPanel";
 import SearchBar from "../components/SearchBar";
 import SkillPanel from "../components/SkillPanel";
 import UserPanel from "../components/UserPanel";
-import { SearchSkillsAndProfilesQuery } from "../generated/graphql";
-import { SEARCH_SKILLS_AND_PROFILES_QUERY } from "../graphql/queries/skills";
+import {
+  GetUserDesireOnEachSkillQuery,
+  SearchSkillsAndProfilesQuery,
+  Skill,
+} from "../generated/graphql";
+import {
+  GET_USER_DESIRE_ON_EACH_SKILL,
+  SEARCH_SKILLS_AND_PROFILES_QUERY,
+} from "../graphql/queries/skills";
 import { i18nContext } from "../utils/i18nContext";
+import CustomSelect from "../components/CustomSelect";
 
 const Search = ({ pathName }) => {
-  const { query } = useRouter();
+  const { t } = useContext(i18nContext);
 
   const isDesktop = useMediaQuery({
     query: "(min-device-width: 1280px)",
   });
-  const { t } = useContext(i18nContext);
+
+  const choicesSorter = [
+    { label: "trendsAsc", value: `${t("search.trends")}` },
+    { label: "default", value: `${t("search.alphabetical")}` },
+    { label: "mostNoted", value: `${t("search.mostNoted")}` },
+  ];
+
   const [search, setSearch] = useState("");
+  const [filter, setFilter] = useState(choicesSorter[0]);
 
   const { data, error } = useQuery<SearchSkillsAndProfilesQuery>(
     SEARCH_SKILLS_AND_PROFILES_QUERY,
     {
       variables: { search: `%${search}%` },
     }
   );
-  if (error) {
-    console.error(error);
-  }
+
+  const { data: responseSkillsDetails, error: skillsDetailsError } =
+    useQuery<GetUserDesireOnEachSkillQuery>(GET_USER_DESIRE_ON_EACH_SKILL, {
+      variables: { search: `%${search}%` },
+    });
+
+  if (error) console.error(error);
+
+  if (skillsDetailsError) console.error(skillsDetailsError);
+
   const skills = data?.skills;
+  const skillsDetails = responseSkillsDetails?.Skill;
   const profiles = data?.profiles;
 
+  const sortedSkills = () => {
```

Comment:
Maybe we can put this code in a useEffect hook. It will prevent from calculating this each time we call sortedSkills

Reply:
Thanks for these suggestions ! Indeed, I think it's better, i'll add it